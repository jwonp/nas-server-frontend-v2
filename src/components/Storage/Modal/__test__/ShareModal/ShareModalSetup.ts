import { afterAll, afterEach, beforeAll } from "vitest";
import { setupServer } from "msw/node";
import { HttpResponse, graphql, http } from "msw";
import { UserSearchResponse } from "@/types/Responses";

const userSearchResponse: UserSearchResponse = {
  searchedUsers: [
    {
      iconURL: "https://placehold.co/600x400/EEE/31343C",
      userId: "useriid",
      username: "박주원",
      email: "user@test.com",
    },
  ],
};

export const restHandlers = [
  http.get(`/api/user/search?query=user@test.com`, () => {
    return HttpResponse.json(userSearchResponse);
  }),
];

const server = setupServer(...restHandlers);

// Start server before all tests
beforeAll(() => server.listen({ onUnhandledRequest: "error" }));

//  Close server after all tests
afterAll(() => server.close());

// Reset handlers after each test `important for test isolation`
afterEach(() => server.resetHandlers());
