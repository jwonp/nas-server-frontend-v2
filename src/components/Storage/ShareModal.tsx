import { useState } from "react";

const ShareModal = () => {
  const [itemName, setItemName] = useState<string>("2019년 아동부 결산");
  return (
    <div>
      <div className="fixed left-0 top-0 w-full h-screen opacity-60 bg-red-600 z-10"></div>
      <div className="fixed left-0 top-0 w-full h-screen z-20">
        <div className="flex">
          <div className="mx-auto w-2/5">
            <div className="flex w-full h-screen">
              <div className="my-auto w-full h-3/5">
                <section className="w-full  h-full bg-slate-500 rounded-lg p-5">
                  <div className="text-xl mb-2">{`"${itemName}" 공유`}</div>
                  <div className="w-full mb-2">
                    <input
                      className="w-full p-2 rounded-lg"
                      type="text"
                      placeholder="공유할 사용자의 이메일 또는 전화번호를 입력하세요."
                    />
                  </div>
                  <section>
                    <div>검색 결과</div>
                    <div>
                      <article></article>
                      <article></article>
                    </div>
                  </section>
                  <section className="flex gap-4 float-right mt-auto">
                    <div className="leading-10">취소</div>
                    <div className="bg-blue-600 py-2 px-5 rounded-xl">
                      <div>공유</div>
                    </div>
                  </section>
                </section>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default ShareModal;
