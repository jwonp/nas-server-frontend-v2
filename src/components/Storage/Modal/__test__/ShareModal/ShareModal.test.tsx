import { expect, test, vi } from "vitest";
import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import ShareModal from "../../ShareModal";
import {
  QueryClient,
  QueryClientProvider,
  useQuery,
} from "@tanstack/react-query";

test("create modal items by search query", async () => {
  const queryClient = new QueryClient();
  const shareModal = render(
    <QueryClientProvider client={queryClient}>
      <ShareModal />
    </QueryClientProvider>
  );
  const searchInput = screen.getByRole("textbox");
  const modalItem = await vi.waitFor(async () => {
    fireEvent.change(searchInput, { target: { value: "user@test.com" } });

    expect((searchInput as HTMLInputElement).value).toBe("user@test.com");
    const modalItem = await screen.findByText("박주원");
    expect(modalItem).toBeDefined();
    return modalItem
  }, 1500);
  expect(screen.getByRole("modal-items")).toBeDefined();
  expect(searchInput).toBeDefined();

  expect(shareModal).toMatchSnapshot();

  cleanup();
});
