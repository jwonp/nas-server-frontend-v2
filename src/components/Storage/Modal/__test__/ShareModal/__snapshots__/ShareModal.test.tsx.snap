// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`create modal items by search query 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="fixed left-0 top-0 w-full h-screen opacity-60 bg-slate-900 z-10"
        />
        <div
          class="fixed left-0 top-0 w-full h-screen z-20"
        >
          <div
            class="flex"
          >
            <div
              class="mx-auto w-2/5 max-md:w-screen"
            >
              <div
                class="flex w-full h-screen"
              >
                <div
                  aria-description="layout"
                  class="my-auto w-full h-4/5 max-md:h-screen  min-h-[512px]"
                >
                  <section
                    class="w-full h-full bg-slate-500 rounded-lg max-md:rounded-none p-5"
                  >
                    <section
                      class="h-20"
                    >
                      <p
                        class="text-xl mb-2"
                      >
                        "2019년 아동부 결산" 공유
                      </p>
                      <article
                        class="w-full mb-2 h-10"
                      >
                        <input
                          class="w-full p-2 rounded-lg text-stone-950 h-full"
                          placeholder="공유할 사용자의 이메일 또는 전화번호를 입력하세요."
                          type="text"
                        />
                      </article>
                    </section>
                    <section
                      class="mb-auto "
                    >
                      <p
                        class="text-xl my-2"
                      >
                        검색 결과
                      </p>
                      <section
                        class="flex gap-2 mb-2 overflow-x-scroll snap-x"
                        role="selected-modal-items"
                      >
                        <article
                          class="flex gap-2 snap-center rounded-full bg-slate-800 min-w-[240px]  h-10 mb-1 py-1 px-2"
                        >
                          <figure
                            class="min-w-[32px] min-h-[32px]"
                          >
                            <img
                              alt=""
                              class="min-w-[32px] min-h-[32px]"
                              data-nimg="1"
                              decoding="async"
                              height="32"
                              loading="lazy"
                              src="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75"
                              srcset="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=32&q=75 1x, /_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75 2x"
                              style="color: transparent;"
                              width="32"
                            />
                          </figure>
                          <p
                            class="leading-8"
                          >
                            tkdel222@gmail.com
                          </p>
                          <div
                            class="cursor-pointer ml-auto py-1 min-w-[24px] min-h-[24px]"
                          >
                            <img
                              alt=""
                              class="hover:bg-slate-600 rounded-full p-1 min-w-[24px] min-h-[24px]"
                              data-nimg="1"
                              decoding="async"
                              height="24"
                              loading="lazy"
                              src="/public/icons/close-white.svg"
                              style="color: transparent;"
                              width="24"
                            />
                          </div>
                        </article>
                        <article
                          class="flex gap-2 snap-center rounded-full bg-slate-800 min-w-[240px]  h-10 mb-1 py-1 px-2"
                        >
                          <figure
                            class="min-w-[32px] min-h-[32px]"
                          >
                            <img
                              alt=""
                              class="min-w-[32px] min-h-[32px]"
                              data-nimg="1"
                              decoding="async"
                              height="32"
                              loading="lazy"
                              src="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75"
                              srcset="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=32&q=75 1x, /_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75 2x"
                              style="color: transparent;"
                              width="32"
                            />
                          </figure>
                          <p
                            class="leading-8"
                          >
                            tkdel222@gmail.com
                          </p>
                          <div
                            class="cursor-pointer ml-auto py-1 min-w-[24px] min-h-[24px]"
                          >
                            <img
                              alt=""
                              class="hover:bg-slate-600 rounded-full p-1 min-w-[24px] min-h-[24px]"
                              data-nimg="1"
                              decoding="async"
                              height="24"
                              loading="lazy"
                              src="/public/icons/close-white.svg"
                              style="color: transparent;"
                              width="24"
                            />
                          </div>
                        </article>
                        <article
                          class="flex gap-2 snap-center rounded-full bg-slate-800 min-w-[240px]  h-10 mb-1 py-1 px-2"
                        >
                          <figure
                            class="min-w-[32px] min-h-[32px]"
                          >
                            <img
                              alt=""
                              class="min-w-[32px] min-h-[32px]"
                              data-nimg="1"
                              decoding="async"
                              height="32"
                              loading="lazy"
                              src="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75"
                              srcset="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=32&q=75 1x, /_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75 2x"
                              style="color: transparent;"
                              width="32"
                            />
                          </figure>
                          <p
                            class="leading-8"
                          >
                            tkdel222@gmail.com
                          </p>
                          <div
                            class="cursor-pointer ml-auto py-1 min-w-[24px] min-h-[24px]"
                          >
                            <img
                              alt=""
                              class="hover:bg-slate-600 rounded-full p-1 min-w-[24px] min-h-[24px]"
                              data-nimg="1"
                              decoding="async"
                              height="24"
                              loading="lazy"
                              src="/public/icons/close-white.svg"
                              style="color: transparent;"
                              width="24"
                            />
                          </div>
                        </article>
                        <article
                          class="flex gap-2 snap-center rounded-full bg-slate-800 min-w-[240px]  h-10 mb-1 py-1 px-2"
                        >
                          <figure
                            class="min-w-[32px] min-h-[32px]"
                          >
                            <img
                              alt=""
                              class="min-w-[32px] min-h-[32px]"
                              data-nimg="1"
                              decoding="async"
                              height="32"
                              loading="lazy"
                              src="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75"
                              srcset="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=32&q=75 1x, /_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75 2x"
                              style="color: transparent;"
                              width="32"
                            />
                          </figure>
                          <p
                            class="leading-8"
                          >
                            tkdel222@gmail.com
                          </p>
                          <div
                            class="cursor-pointer ml-auto py-1 min-w-[24px] min-h-[24px]"
                          >
                            <img
                              alt=""
                              class="hover:bg-slate-600 rounded-full p-1 min-w-[24px] min-h-[24px]"
                              data-nimg="1"
                              decoding="async"
                              height="24"
                              loading="lazy"
                              src="/public/icons/close-white.svg"
                              style="color: transparent;"
                              width="24"
                            />
                          </div>
                        </article>
                        <article
                          class="flex gap-2 snap-center rounded-full bg-slate-800 min-w-[240px]  h-10 mb-1 py-1 px-2"
                        >
                          <figure
                            class="min-w-[32px] min-h-[32px]"
                          >
                            <img
                              alt=""
                              class="min-w-[32px] min-h-[32px]"
                              data-nimg="1"
                              decoding="async"
                              height="32"
                              loading="lazy"
                              src="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75"
                              srcset="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=32&q=75 1x, /_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75 2x"
                              style="color: transparent;"
                              width="32"
                            />
                          </figure>
                          <p
                            class="leading-8"
                          >
                            tkdel222@gmail.com
                          </p>
                          <div
                            class="cursor-pointer ml-auto py-1 min-w-[24px] min-h-[24px]"
                          >
                            <img
                              alt=""
                              class="hover:bg-slate-600 rounded-full p-1 min-w-[24px] min-h-[24px]"
                              data-nimg="1"
                              decoding="async"
                              height="24"
                              loading="lazy"
                              src="/public/icons/close-white.svg"
                              style="color: transparent;"
                              width="24"
                            />
                          </div>
                        </article>
                        <article
                          class="flex gap-2 snap-center rounded-full bg-slate-800 min-w-[240px]  h-10 mb-1 py-1 px-2"
                        >
                          <figure
                            class="min-w-[32px] min-h-[32px]"
                          >
                            <img
                              alt=""
                              class="min-w-[32px] min-h-[32px]"
                              data-nimg="1"
                              decoding="async"
                              height="32"
                              loading="lazy"
                              src="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75"
                              srcset="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=32&q=75 1x, /_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75 2x"
                              style="color: transparent;"
                              width="32"
                            />
                          </figure>
                          <p
                            class="leading-8"
                          >
                            tkdel222@gmail.com
                          </p>
                          <div
                            class="cursor-pointer ml-auto py-1 min-w-[24px] min-h-[24px]"
                          >
                            <img
                              alt=""
                              class="hover:bg-slate-600 rounded-full p-1 min-w-[24px] min-h-[24px]"
                              data-nimg="1"
                              decoding="async"
                              height="24"
                              loading="lazy"
                              src="/public/icons/close-white.svg"
                              style="color: transparent;"
                              width="24"
                            />
                          </div>
                        </article>
                        <article
                          class="flex gap-2 snap-center rounded-full bg-slate-800 min-w-[240px]  h-10 mb-1 py-1 px-2"
                        >
                          <figure
                            class="min-w-[32px] min-h-[32px]"
                          >
                            <img
                              alt=""
                              class="min-w-[32px] min-h-[32px]"
                              data-nimg="1"
                              decoding="async"
                              height="32"
                              loading="lazy"
                              src="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75"
                              srcset="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=32&q=75 1x, /_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75 2x"
                              style="color: transparent;"
                              width="32"
                            />
                          </figure>
                          <p
                            class="leading-8"
                          >
                            tkdel222@gmail.com
                          </p>
                          <div
                            class="cursor-pointer ml-auto py-1 min-w-[24px] min-h-[24px]"
                          >
                            <img
                              alt=""
                              class="hover:bg-slate-600 rounded-full p-1 min-w-[24px] min-h-[24px]"
                              data-nimg="1"
                              decoding="async"
                              height="24"
                              loading="lazy"
                              src="/public/icons/close-white.svg"
                              style="color: transparent;"
                              width="24"
                            />
                          </div>
                        </article>
                        <article
                          class="flex gap-2 snap-center rounded-full bg-slate-800 min-w-[240px]  h-10 mb-1 py-1 px-2"
                        >
                          <figure
                            class="min-w-[32px] min-h-[32px]"
                          >
                            <img
                              alt=""
                              class="min-w-[32px] min-h-[32px]"
                              data-nimg="1"
                              decoding="async"
                              height="32"
                              loading="lazy"
                              src="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75"
                              srcset="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=32&q=75 1x, /_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75 2x"
                              style="color: transparent;"
                              width="32"
                            />
                          </figure>
                          <p
                            class="leading-8"
                          >
                            tkdel222@gmail.com
                          </p>
                          <div
                            class="cursor-pointer ml-auto py-1 min-w-[24px] min-h-[24px]"
                          >
                            <img
                              alt=""
                              class="hover:bg-slate-600 rounded-full p-1 min-w-[24px] min-h-[24px]"
                              data-nimg="1"
                              decoding="async"
                              height="24"
                              loading="lazy"
                              src="/public/icons/close-white.svg"
                              style="color: transparent;"
                              width="24"
                            />
                          </div>
                        </article>
                      </section>
                      <section
                        class="col-span-1 mb-3 overflow-y-scroll h-32 select-none"
                        role="modal-items"
                      />
                    </section>
                    <section
                      class="float-right h-10"
                    >
                      <article
                        class="flex gap-4"
                      >
                        <button
                          class="leading-10"
                        >
                          취소
                        </button>
                        <button
                          class="bg-blue-600 py-2 px-5 rounded-xl"
                        >
                          공유
                        </button>
                      </article>
                    </section>
                  </section>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="fixed left-0 top-0 w-full h-screen opacity-60 bg-slate-900 z-10"
      />
      <div
        class="fixed left-0 top-0 w-full h-screen z-20"
      >
        <div
          class="flex"
        >
          <div
            class="mx-auto w-2/5 max-md:w-screen"
          >
            <div
              class="flex w-full h-screen"
            >
              <div
                aria-description="layout"
                class="my-auto w-full h-4/5 max-md:h-screen  min-h-[512px]"
              >
                <section
                  class="w-full h-full bg-slate-500 rounded-lg max-md:rounded-none p-5"
                >
                  <section
                    class="h-20"
                  >
                    <p
                      class="text-xl mb-2"
                    >
                      "2019년 아동부 결산" 공유
                    </p>
                    <article
                      class="w-full mb-2 h-10"
                    >
                      <input
                        class="w-full p-2 rounded-lg text-stone-950 h-full"
                        placeholder="공유할 사용자의 이메일 또는 전화번호를 입력하세요."
                        type="text"
                      />
                    </article>
                  </section>
                  <section
                    class="mb-auto "
                  >
                    <p
                      class="text-xl my-2"
                    >
                      검색 결과
                    </p>
                    <section
                      class="flex gap-2 mb-2 overflow-x-scroll snap-x"
                      role="selected-modal-items"
                    >
                      <article
                        class="flex gap-2 snap-center rounded-full bg-slate-800 min-w-[240px]  h-10 mb-1 py-1 px-2"
                      >
                        <figure
                          class="min-w-[32px] min-h-[32px]"
                        >
                          <img
                            alt=""
                            class="min-w-[32px] min-h-[32px]"
                            data-nimg="1"
                            decoding="async"
                            height="32"
                            loading="lazy"
                            src="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75"
                            srcset="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=32&q=75 1x, /_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75 2x"
                            style="color: transparent;"
                            width="32"
                          />
                        </figure>
                        <p
                          class="leading-8"
                        >
                          tkdel222@gmail.com
                        </p>
                        <div
                          class="cursor-pointer ml-auto py-1 min-w-[24px] min-h-[24px]"
                        >
                          <img
                            alt=""
                            class="hover:bg-slate-600 rounded-full p-1 min-w-[24px] min-h-[24px]"
                            data-nimg="1"
                            decoding="async"
                            height="24"
                            loading="lazy"
                            src="/public/icons/close-white.svg"
                            style="color: transparent;"
                            width="24"
                          />
                        </div>
                      </article>
                      <article
                        class="flex gap-2 snap-center rounded-full bg-slate-800 min-w-[240px]  h-10 mb-1 py-1 px-2"
                      >
                        <figure
                          class="min-w-[32px] min-h-[32px]"
                        >
                          <img
                            alt=""
                            class="min-w-[32px] min-h-[32px]"
                            data-nimg="1"
                            decoding="async"
                            height="32"
                            loading="lazy"
                            src="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75"
                            srcset="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=32&q=75 1x, /_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75 2x"
                            style="color: transparent;"
                            width="32"
                          />
                        </figure>
                        <p
                          class="leading-8"
                        >
                          tkdel222@gmail.com
                        </p>
                        <div
                          class="cursor-pointer ml-auto py-1 min-w-[24px] min-h-[24px]"
                        >
                          <img
                            alt=""
                            class="hover:bg-slate-600 rounded-full p-1 min-w-[24px] min-h-[24px]"
                            data-nimg="1"
                            decoding="async"
                            height="24"
                            loading="lazy"
                            src="/public/icons/close-white.svg"
                            style="color: transparent;"
                            width="24"
                          />
                        </div>
                      </article>
                      <article
                        class="flex gap-2 snap-center rounded-full bg-slate-800 min-w-[240px]  h-10 mb-1 py-1 px-2"
                      >
                        <figure
                          class="min-w-[32px] min-h-[32px]"
                        >
                          <img
                            alt=""
                            class="min-w-[32px] min-h-[32px]"
                            data-nimg="1"
                            decoding="async"
                            height="32"
                            loading="lazy"
                            src="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75"
                            srcset="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=32&q=75 1x, /_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75 2x"
                            style="color: transparent;"
                            width="32"
                          />
                        </figure>
                        <p
                          class="leading-8"
                        >
                          tkdel222@gmail.com
                        </p>
                        <div
                          class="cursor-pointer ml-auto py-1 min-w-[24px] min-h-[24px]"
                        >
                          <img
                            alt=""
                            class="hover:bg-slate-600 rounded-full p-1 min-w-[24px] min-h-[24px]"
                            data-nimg="1"
                            decoding="async"
                            height="24"
                            loading="lazy"
                            src="/public/icons/close-white.svg"
                            style="color: transparent;"
                            width="24"
                          />
                        </div>
                      </article>
                      <article
                        class="flex gap-2 snap-center rounded-full bg-slate-800 min-w-[240px]  h-10 mb-1 py-1 px-2"
                      >
                        <figure
                          class="min-w-[32px] min-h-[32px]"
                        >
                          <img
                            alt=""
                            class="min-w-[32px] min-h-[32px]"
                            data-nimg="1"
                            decoding="async"
                            height="32"
                            loading="lazy"
                            src="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75"
                            srcset="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=32&q=75 1x, /_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75 2x"
                            style="color: transparent;"
                            width="32"
                          />
                        </figure>
                        <p
                          class="leading-8"
                        >
                          tkdel222@gmail.com
                        </p>
                        <div
                          class="cursor-pointer ml-auto py-1 min-w-[24px] min-h-[24px]"
                        >
                          <img
                            alt=""
                            class="hover:bg-slate-600 rounded-full p-1 min-w-[24px] min-h-[24px]"
                            data-nimg="1"
                            decoding="async"
                            height="24"
                            loading="lazy"
                            src="/public/icons/close-white.svg"
                            style="color: transparent;"
                            width="24"
                          />
                        </div>
                      </article>
                      <article
                        class="flex gap-2 snap-center rounded-full bg-slate-800 min-w-[240px]  h-10 mb-1 py-1 px-2"
                      >
                        <figure
                          class="min-w-[32px] min-h-[32px]"
                        >
                          <img
                            alt=""
                            class="min-w-[32px] min-h-[32px]"
                            data-nimg="1"
                            decoding="async"
                            height="32"
                            loading="lazy"
                            src="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75"
                            srcset="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=32&q=75 1x, /_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75 2x"
                            style="color: transparent;"
                            width="32"
                          />
                        </figure>
                        <p
                          class="leading-8"
                        >
                          tkdel222@gmail.com
                        </p>
                        <div
                          class="cursor-pointer ml-auto py-1 min-w-[24px] min-h-[24px]"
                        >
                          <img
                            alt=""
                            class="hover:bg-slate-600 rounded-full p-1 min-w-[24px] min-h-[24px]"
                            data-nimg="1"
                            decoding="async"
                            height="24"
                            loading="lazy"
                            src="/public/icons/close-white.svg"
                            style="color: transparent;"
                            width="24"
                          />
                        </div>
                      </article>
                      <article
                        class="flex gap-2 snap-center rounded-full bg-slate-800 min-w-[240px]  h-10 mb-1 py-1 px-2"
                      >
                        <figure
                          class="min-w-[32px] min-h-[32px]"
                        >
                          <img
                            alt=""
                            class="min-w-[32px] min-h-[32px]"
                            data-nimg="1"
                            decoding="async"
                            height="32"
                            loading="lazy"
                            src="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75"
                            srcset="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=32&q=75 1x, /_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75 2x"
                            style="color: transparent;"
                            width="32"
                          />
                        </figure>
                        <p
                          class="leading-8"
                        >
                          tkdel222@gmail.com
                        </p>
                        <div
                          class="cursor-pointer ml-auto py-1 min-w-[24px] min-h-[24px]"
                        >
                          <img
                            alt=""
                            class="hover:bg-slate-600 rounded-full p-1 min-w-[24px] min-h-[24px]"
                            data-nimg="1"
                            decoding="async"
                            height="24"
                            loading="lazy"
                            src="/public/icons/close-white.svg"
                            style="color: transparent;"
                            width="24"
                          />
                        </div>
                      </article>
                      <article
                        class="flex gap-2 snap-center rounded-full bg-slate-800 min-w-[240px]  h-10 mb-1 py-1 px-2"
                      >
                        <figure
                          class="min-w-[32px] min-h-[32px]"
                        >
                          <img
                            alt=""
                            class="min-w-[32px] min-h-[32px]"
                            data-nimg="1"
                            decoding="async"
                            height="32"
                            loading="lazy"
                            src="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75"
                            srcset="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=32&q=75 1x, /_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75 2x"
                            style="color: transparent;"
                            width="32"
                          />
                        </figure>
                        <p
                          class="leading-8"
                        >
                          tkdel222@gmail.com
                        </p>
                        <div
                          class="cursor-pointer ml-auto py-1 min-w-[24px] min-h-[24px]"
                        >
                          <img
                            alt=""
                            class="hover:bg-slate-600 rounded-full p-1 min-w-[24px] min-h-[24px]"
                            data-nimg="1"
                            decoding="async"
                            height="24"
                            loading="lazy"
                            src="/public/icons/close-white.svg"
                            style="color: transparent;"
                            width="24"
                          />
                        </div>
                      </article>
                      <article
                        class="flex gap-2 snap-center rounded-full bg-slate-800 min-w-[240px]  h-10 mb-1 py-1 px-2"
                      >
                        <figure
                          class="min-w-[32px] min-h-[32px]"
                        >
                          <img
                            alt=""
                            class="min-w-[32px] min-h-[32px]"
                            data-nimg="1"
                            decoding="async"
                            height="32"
                            loading="lazy"
                            src="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75"
                            srcset="/_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=32&q=75 1x, /_next/image?url=%2Fpublic%2Ficons%2FuserCircle.png&w=64&q=75 2x"
                            style="color: transparent;"
                            width="32"
                          />
                        </figure>
                        <p
                          class="leading-8"
                        >
                          tkdel222@gmail.com
                        </p>
                        <div
                          class="cursor-pointer ml-auto py-1 min-w-[24px] min-h-[24px]"
                        >
                          <img
                            alt=""
                            class="hover:bg-slate-600 rounded-full p-1 min-w-[24px] min-h-[24px]"
                            data-nimg="1"
                            decoding="async"
                            height="24"
                            loading="lazy"
                            src="/public/icons/close-white.svg"
                            style="color: transparent;"
                            width="24"
                          />
                        </div>
                      </article>
                    </section>
                    <section
                      class="col-span-1 mb-3 overflow-y-scroll h-32 select-none"
                      role="modal-items"
                    />
                  </section>
                  <section
                    class="float-right h-10"
                  >
                    <article
                      class="flex gap-4"
                    >
                      <button
                        class="leading-10"
                      >
                        취소
                      </button>
                      <button
                        class="bg-blue-600 py-2 px-5 rounded-xl"
                      >
                        공유
                      </button>
                    </article>
                  </section>
                </section>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`create modal items by search query 2`] = `
<section
  class="col-span-1 mb-3 overflow-y-scroll h-32 select-none"
  role="modal-items"
/>
`;
